[게시판 회원 기능] 베디 미션 제출 합니다.

dpudpu opened the pull request:
현구님 안녕하세요. 
리뷰를 위해서 귀한 시간 내주셔서 감사합니다.

몇 가지 질문이 있습니다.

1. 제가 TDD를 하려고 노력했는데, Repository, Entity 같은 경우에는 비즈니스 로직이 없으면 어떻게 해야 할까요?  test 코드에서 builder나 javabean 테스트하면서 생성해주면 될까요? 아니면 프로덕션 코드 작성 후에 나중에 필요할 때 테스트를 작성하면 될까요?
2. 각 컨트롤러의 URI를 상수처리 했는데, 이렇게 사용하는 것에 대해서 어떻게 생각하시나요? 
3. 테스트 코드 중복제거가 아주 어렵더라고요. 제거한다고 오히려 가독성이 더 떨어지는 거 같고, 혹시 조언 좀 받을 수 있을까요?

리뷰해주셔서 감사합니다.

vsh123 committed:
remove : 기존 로직 삭제

vsh123 committed:
feat : MainController 구현

vsh123 committed:
feat: Article 구현

vsh123 committed:
feat: ArticleReposiory 구현

vsh123 committed:
feat: ArticleController.write()

vsh123 committed:
feat: ArticleController.show()

vsh123 committed:
feat: ArticleController.editForm()

vsh123 committed:
ArticleController.edit()

vsh123 committed:
feat: ArticleController.delete()

vsh123 committed:
feat: ArticleController.writeForm()

vsh123 committed:
rename: ArticleControllerTests

vsh123 committed:
feat: MainController Aritcles 조회

vsh123 committed:
refactor: @transactional 추가

vsh123 committed:
rename: template 이름 변경

vsh123 committed:
feat: UserController.signupForm()

vsh123 committed:
feat: 회원가입 유효성 검사

vsh123 committed:
feat: 회원가입 이메일 중복, 패스워드 중복 검사

vsh123 committed:
feat: 회원가입 프론트 구현

vsh123 committed:
feat: 로그인 구현

vsh123 committed:
feat: login 구현

vsh123 committed:
feat: 로그아웃

vsh123 committed:
feat: 회원 목록 조회

vsh123 committed:
feat: 마이페이지, 수정폼 구현

vsh123 committed:
feat: 회원정보 수정

vsh123 committed:
feat: 회원탈퇴

vsh123 committed:
refactor: UserDto

vsh123 committed:
refactor: 테스트코드 중복제거

vsh123 committed:
feat: interceptor

dpudpu committed:
feat: logger 추가

dpudpu committed:
test: LoginInterceptor 세션 테스트 추가

dpudpu committed:
feat: jupiter-params 추가

dpudpu committed:
refactor: html 중복 제거

dpudpu committed:
feat: Articles Paging 구현

dpudpu committed:
test: UserControllerTest 세션 테스트 추가

dpudpu committed:
refactor: uri 상수 처리

dpudpu committed:
test: UserServiceTest

dpudpu committed:
style: 포맷팅

dpudpu committed:
refactor: 샘플데이터 수정

dpudpu committed:
refactor: user session 상수 처리

dpudpu committed:
rename: ValidSignupException -> ValidUserException

dpudpu committed:
refactor: LoginInterceptor 매직넘버 상수처리

dpudpu committed:
fix: logout AuthInterceptor에서 튕기는 문제 해결

kang-hyungu commented:
1. repository 구현체는 이미 외부에서 구현된 코드니 굳이 테스트할 필요가 없다고 생각합니다. tdd할 때 repository는 이미 구현된 코드라 생각하고 작성하시면 될거에요. entity는 비즈니스 로직이 없을 정도로 간단한 엔티티면 따로 테스트할 필요는 없겠죠.
그런데 코드를 보니 user 유효성 체크를 userDto에서 하고 있네요. 사용자 이름이 2~10자이거나 비밀번호가 8자리인 조건들은 사실 비즈니스 로직에 해당되지 않을까요? 유효성 체크를 도메인 계층으로 옮기면 테스트해볼 사항이 생길거에요.
2. uri를 상수로 관리했을 때 장점이 없는 것 같네요.
3. webClient를 사용할 때 중복되는 코드들은 BaseControllerTest 같은 슈퍼 클래스를 만들고 UserControllerTest에서 상속 받아서 슈퍼 클래스에 이미 구현된 메서드들을 활용하는 방법도 있어요

kang-hyungu reviewed (changes requested):
안녕하세요 베디님 리뷰어 강현구입니다.
스프링을 사용해보신 경험이 있나보네요
능숙하게 잘 구현하셨어요 :+1: 
몇 가지 피드백 남겼으니 확인해보시고 반영해보아요

dpudpu committed:
style: build.gradle 정리

dpudpu committed:
refactor: errorAttributes() 를 ErrorConfig.class 로 분리

dpudpu committed:
refactor: uri 상수 제거

dpudpu committed:
refactor: Article.update() 수정

dpudpu committed:
refactor: test 코드 필요 없는 어노테이션 제거

dpudpu committed:
refactor: 인터셉터 클래스 commons 패키지에서 interceptor 로 변경

dpudpu committed:
refactor: BeanUtils 제거, User setter 제거

dpudpu committed:
refactor: MainController.index Pageable 파라미터로 변경

dpudpu committed:
feat: User.authenticate 추가

dpudpu committed:
refactor: ControllerTests 에서 service 빈 제거하고 api를 직접 호출하는 방식으로 변경

dpudpu committed:
refactor: user session 처리용 UserSession 생성

dpudpu committed:
style: 포맷팅

dpudpu commented:
현구님 리뷰 해주셔서 감사합니다. 
피드백 해주신 부분들 수정하면서 몇가지 질문들 코멘트에 달았는데 확인해주시면 감사하겠습니다.
그리고 도메인에도 유효성 검사 추가했습니다. 이 부분 어떤지 말씀해주시면 감사하겠습니다.


감사합니다.

dpudpu committed:
feat: User 유효성검사 추가

dpudpu committed:
refactor: Article @Builder 방식 변경

dpudpu committed:
fix: LoginInterceptor 적용 패스 및 테스트코드 추가

dpudpu committed:
fix: UserDto.Update.toUser()

UserDto.Update.toUser() 에서 User를 생성할 때 유효성검사 때문에 생성 안되던 문제 해결

dpudpu committed:
fix: test코드 session 추가

dpudpu committed:
docs: application.properties spring.jpa.show-sql=true 제거

dpudpu committed:
refactor: UserDto.Update.toUser() 방식 수정

dpudpu committed:
refactor: UserDto.Update에 id 필드 추가

dpudpu committed:
refactor: User unique 설정 방식 변경

alter table user add constraint email unique (email)
@Column으로 이름 설정하면 이름이 이상하게 설정된다.
@Table(uniqueConstraints = @UniqueConstraint(name = "email", columnNames = {"email"}))
를 사용하면 이름이 제대로 표시된다

dpudpu committed:
refactor: SessionId 상수 UserSession 으로 이동

dpudpu committed:
refactor: UserService.login() UserSession 반환 하는 부분 수정

UserSession은 DTO가 아니므로 Controller에서 생성되게 변경

dpudpu committed:
feat: UserSessionArgumentResolver

dpudpu committed:
refactor: Entity 기본생성자 private으로 변경

kang-hyungu commented:
user 도메인 클래스에 유효성 검사 추가한 내용 확인했어요!
조금만 더 수정하면 좋을 것 같아요
static final로 선언한 유효성 검사용 상수들이 User 객체 전체에서 관심가져야 할 내용일까요?
각 validateXXX 메서드에서 알고 있으면 충분한 내용 아닐까요?

실제 user 객체를 이루는 속성보다 유효성 검사용 상수가 많아서 객체 파악할 때 헷갈릴 것 같아요

kang-hyungu reviewed (approved):
피드백 반영하신 부분 확인했어요!
그리고 질문 주신 내용들에 대해서 제 의견도 남겼어요
꼭 반영할 필요는 없으니 개발하실 때 참고만 하세요
추가로 수정하고 싶으신 부분은 다음 단계 진행하시면서 적용하셔도 충분해보이니 머지 처리할게요
코드 구현하느라 수고하셨어요!
